src/services/resource.js

```diff
@@ -8,196 +8,580 @@
+				result,
+			})
+		}
+		const uniqueResourceIds = resourcesCreatedByMe.map((item) => item.resource_id)
 
-				filter.organization_id = {
+		// get the unique organization ids from resource creator mapping table by the user
+		const OrganizationIds = utils.getUniqueElements(resourcesCreatedByMe.map((item) => item.organization_id))
+
+		const filter = await this.constructCustomFilter(
+			{
+				user_id: loggedInUserId,
+				organization_id: {
 					[Op.in]: OrganizationIds,
-				}
-				filter.status = {
-					[Op.in]: common.PAGE_STATUS_VALUES[queryParams[common.PAGE_STATUS]],
+				},
+				id: {
+					[Op.in]: uniqueResourceIds,
+				},
+				status: {
+					[Op.in]: common.PAGE_STATUS_VALUES[common.PAGE_STATUS_DRAFTS],
+				},
+			},
+			queryParams,
+			searchText
+		)
+
+		const sort = await this.constructSortOptions(queryParams)
+		const response = await resourceQueries.resourceList(
+			filter,
+			['id', 'title', 'organization_id', 'type', 'status', 'user_id', 'created_at', 'updated_at'],
+			sort,
+			page,
+			limit
+		)
+		if (response.result.length <= 0) {
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'RESOURCE_LISTED_SUCCESSFULLY',
+				result,
+			})
+		}
+
+		const userDetails = await this.fetchUserDetails(
+			utils.getUniqueElements(response.result.map((item) => item.user_id))
+		)
+		const orgDetails = await this.fetchOrganizationDetails(
+			utils.getUniqueElements(response.result.map((item) => item.organization_id))
+		)
+		result = await this.responseBuilder(response, userDetails, orgDetails, {})
+
+		return responses.successResponse({
+			statusCode: httpStatusCode.ok,
+			message: 'RESOURCE_LISTED_SUCCESSFULLY',
+			result,
+		})
+	}
+
+	// build common response for the functions.
+	// it accepts the resource details , user details obj , org details obj , and additional details based on business logic
+	static async responseBuilder(resourceDetails, userDetails, orgDetails, additionalResourceInformation) {
+		let result = {}
+
+		result.data = resourceDetails.result.map((res) => {
+			res.organization = orgDetails[res.organization_id] ? orgDetails[res.organization_id] : {}
+			res.creator = userDetails[res.user_id] && userDetails[res.user_id].name ? userDetails[res.user_id].name : ''
+			res.notes = res.meta.notes ? res.meta.notes : ''
+
+			if (additionalResourceInformation[res.id]) {
+				res = {
+					...res,
+					...additionalResourceInformation[res.id],
 				}
 			}
+			delete res.user_id
+			delete res.organization_id
+			delete res.meta
+			return res
+		})
+
+		result.count = resourceDetails.count
+		return result
+	}
 
-			if (queryParams[common.PAGE_STATUS] === common.PAGE_STATUS_SUBMITTED_FOR_REVIEW) {
-				// specific filters for submitted for review page
+	// custom construction of filter from all the given params
+	static async constructCustomFilter(filter, queryParams, searchText) {
+		let returnFilter = {
+			...filter,
+		}
+		if (searchText.length > 0) {
```

Comment:
@adithyadinesh0412  searchText is a string right ?
If its string then no need to check length